packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -68,15 +66,28 @@ class TransactionService {
     this.currentTransaction = value
   }
 
-  startCounter(transaction) {
-    transaction.browserResponsivenessCounter = 0
-    const id = setInterval(function() {
-      if (transaction.ended) {
-        window.clearInterval(id)
-      } else {
-        transaction.browserResponsivenessCounter++
+  ensureRespInterval() {
+    let currentTr = this.getCurrentTransaction()
```

Comment:
This is double work, When we call `ensureRespInterval`, we have already created a current transaction, So instead of again getting the current transaction, we can simply pass that flag in args. 

```js
ensureRespInterval(checkBrowserResponsiveness) {
  if (!checkBrowserResponsiveness) {
     return
   }
 // code here
}

```

Reply:
Let's move `checkBrowserResponsiveness` to the caller function!